homeassistant/components/lifx/util.py

```diff
@@ -0,0 +1,162 @@
+    """
+    hue, saturation, brightness, kelvin = [None] * 4
+
+    preprocess_turn_on_alternatives(hass, kwargs)  # type: ignore[no-untyped-call]
+
+    if ATTR_HS_COLOR in kwargs:
+        hue, saturation = kwargs[ATTR_HS_COLOR]
+    elif ATTR_RGB_COLOR in kwargs:
+        hue, saturation = color_util.color_RGB_to_hs(*kwargs[ATTR_RGB_COLOR])
+    elif ATTR_XY_COLOR in kwargs:
+        hue, saturation = color_util.color_xy_to_hs(*kwargs[ATTR_XY_COLOR])
+
+    if hue is not None:
+        assert saturation is not None
+        hue = int(hue / 360 * 65535)
+        saturation = int(saturation / 100 * 65535)
+        kelvin = 3500
+
+    if ATTR_COLOR_TEMP in kwargs:
+        kelvin = int(
+            color_util.color_temperature_mired_to_kelvin(kwargs[ATTR_COLOR_TEMP])
+        )
+        saturation = 0
+
+    if ATTR_BRIGHTNESS in kwargs:
+        brightness = convert_8_to_16(kwargs[ATTR_BRIGHTNESS])
+
+    hsbk = [hue, saturation, brightness, kelvin]
+    return None if hsbk == [None] * 4 else hsbk
+
+
+def merge_hsbk(
+    base: list[float | int | None], change: list[float | int | None]
+) -> list[float | int | None]:
+    """Copy change on top of base, except when None.
+
+    Hue, Saturation, Brightness, Kelvin
+    """
+    return [b if c is None else c for b, c in zip(base, change)]
+
+
+def _get_mac_offset(mac_addr: str, offset: int) -> str:
+    octets = [int(octet, 16) for octet in mac_addr.split(":")]
+    octets[5] = (octets[5] + offset) % 256
+    return ":".join(f"{octet:02x}" for octet in octets)
+
+
+def _off_by_one_mac(firmware: str) -> bool:
+    """Check if the firmware version has the off by one mac."""
+    return bool(firmware and AwesomeVersion(firmware) >= FIX_MAC_FW)
+
+
+def get_real_mac_addr(mac_addr: str, firmware: str) -> str:
+    """Increment the last byte of the mac address by one for FW>3.70."""
+    return _get_mac_offset(mac_addr, 1) if _off_by_one_mac(firmware) else mac_addr
+
+
+def formatted_serial(serial_number: str) -> str:
+    """Format the serial number to match the HA device registry."""
+    return dr.format_mac(serial_number)
+
+
+def mac_matches_serial_number(mac_addr: str, serial_number: str) -> bool:
+    """Check if a mac address matches the serial number."""
+    formatted_mac = dr.format_mac(mac_addr)
+    return bool(
+        formatted_serial(serial_number) == formatted_mac
+        or _get_mac_offset(serial_number, 1) == formatted_mac
+    )
+
+
+async def async_execute_lifx(method: Callable) -> Message:
+    """Execute a lifx coroutine and wait for a response."""
+    future: asyncio.Future[Message] = asyncio.Future()
+
+    def _callback(bulb: Light, message: Message) -> None:
+        future.set_result(message)
+
+    _LOGGER.debug("Sending LIFX command: %s", method)
+
+    method(callb=_callback)
+    result = None
+
+    async with async_timeout.timeout(OVERALL_TIMEOUT):
+        result = await future
+
+    _LOGGER.debug("LIFX command result: %s", result)
```

Comment:
This is extremely noisy and I'm not convinced it's anything but noise for troubleshooting post-release. We probably don't need the entire response object in the logs for that.

Reply:
removed via [`35e965a` (#74316)](https://github.com/home-assistant/core/pull/74316/commits/35e965a046870b64b3d307980f540cc3587980d2)